great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -300,11 +300,14 @@ def __init__(
         # initialize runtime_environment as empty dict if None
         runtime_environment = runtime_environment or {}
         if self._ge_cloud_mode:
+            assert ge_cloud_config is not None
```

Comment:
How does this assertion square with the `if project_config is None` conditional in the `CloudDataContext` constructor? It seems like we have a way to access a default config or something based on `CloudDataConext.retrieve_data_context_config_from_ge_cloud`. Maybe I'm not understanding something, but why would we want to be strict about raising an error here?

Reply:
I see now that we have both `project_config` and `ge_cloud_config`. What is the difference?